test/OpenTelemetry.Tests/Logs/OpenTelemetryLoggingExtensionsTests.cs

```diff
@@ -114,4 +115,18 @@ void OptionsCallback(OpenTelemetryLoggerOptions options)
             optionsCallbackInvocations++;
         }
     }
+
+    // This test validates that the OpenTelemetryLoggerOptions contains only primitive type properties.
+    // This is necessary to ensure trim correctness since that class is effectively deserialized from
+    // configuration. The top level properties are ensured via annotation on the RegisterProviderOptions API
+    // but if there was any complex type property, members of the complex type would not be preserved
+    // and could lead to incompatibilities with trimming.
+    [Fact]
+    public void TestTrimmingCorrectnessOfOpenTelemetryLoggerOptions()
+    {
+        foreach (var prop in typeof(OpenTelemetryLoggerOptions).GetProperties(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance))
```

Comment:
just curious - why test only these 3 BindingFlags?

Reply:
These 3 flags basically mean "everything but static properties". Configuration binder doesn't bind static members, so we can ignore those.